test_msgs/CMakeLists.txt

```diff
@@ -40,14 +47,14 @@ if(BUILD_TESTING)
   target_include_directories(test_action_typesupport_c_builds PRIVATE
     include)
   target_link_libraries(test_action_typesupport_c_builds
-    ${_AMENT_EXPORT_LIBRARIES})
+    generate_test_msgs__rosidl_typesupport_c)
```

Comment:
@sloretz hmm, I'm a bit uncomfortable depending on targets that may or may not be available (as these are provided by `rosidl`-specific `ament` extensions). I know I'm opening a can of worms but if we have to do this, something's off.

Reply:
Yeah, I think there isn't good support for using generated messages from the same package that generates them. It looks like a better option would be to use `rosidl_get_typesupport_target()` which was at least added for this purpose: https://github.com/ros2/rosidl/pull/606

Reply:
Using `rosidl_get_typesupport_target()` in 49addc7